### DISCUSSION TITLE
Is there a way to for a stackstorm action / alias to output text to a text snippet in slack?

### DISCUSSION THREAD

**Discussion by planet** [StackStorm/st2#5979] [ChatOps]
As you all already slack doesn't handle large amounts of text well.  

Is there a way to for a stackstorm action / alias to output text to a text snippet in slack?

If so are there any guides on how to do this in the stackstorm alias?

**Comment by nzlosh** [comment]
Notifications https://docs.stackstorm.com/chatops/notifications.html are supported and it's possible to use jinja templates for formatting the message.

**Comment by planet** [comment]
@nzlosh not sure I understand how to use notifications to post a text snippet in slack.  Can you provide some more detail or example on how to accomplish this?  

So basically this is what I'm looking for:

1. Someone sends a chatops command to the stackstorm chatops bot. 
2. The bot runs a local shell cmd
3. The output of the script gets returned in a text snippet to the person either through a DM or via the channel depending where the command was triggered from.

**Comment by nzlosh** [comment]
If Slack snippets are some sort of special slack formatting, then you'll need to configure the action alias to send it.  Unfortunately, I don't have any examples or more exact instructions.

As a general guide, StackStorm has ways to pass extra parameters for Slack, see the documentation https://docs.stackstorm.com/chatops/aliases.html.  These extra parameters are defined by the Slack API specification.  To know the structures, look at Slack's API documentation https://api.slack.com/messaging/composing/layouts

**Comment by planet** [comment]
@nzlosh figured out how to work around this by having the script post to slack instead. Related question what is the variable I can use to grab the username or channel on how the chatops alias was triggered from?

I know the variable `{{ execution.context.api_user }}` will give you the slack username after running it but it is possible to pass the variable and run an action with this username as an parameter?

**Comment by planet** [comment]
Nevermind I figured it out.  All good.
